[33기 김형겸] ADD: Social Login - Kakao Login API 구현

Kyeom1997 opened the pull request:
1. Kakao Login API 활용하여 Redirect URI로 백엔드에 client id 전달

## :: 최근 작업 주제 (하나 이상의 주제를 선택해주세요.)
- [x] 기능 추가
- [x] 리뷰 반영
- [ ] 리팩토링
- [ ] 버그 수정
- [ ] 컨벤션 수정

<br />

## :: 구현 목표 (해당 브랜치에서 구현하고자 하는 하나의 목표를 설정합니다.)
- 카카오, 깃허브, 페이스북, 네이버, 구글 오픈소스 API를 통한 소셜 로그인 구현

<br />

## :: 구현 사항 설명 (작업한 내용을 상세하게 기록합니다.)
1. Kakao Login API를 통한 Redirect URI 백엔드에 전달
2. Redirect 컴포넌트 구현 완료 및 token setItem시 main Component로 이동 구현
3. CLIENT_ID, REDIRECT_URI, KAKAO_AUTH_URL 별도의 파일에 분리하여 관리

4. 

<br />

## :: 성장 포인트 (해당 기능을 구현하며 고민했던 사항이나 새로 알게된 부분, 어려웠던 점 등을 작성합니다.)


<br />

## :: 기타 질문 및 특이 사항
현재 카카오 로그인만 구현한 상태라서 map을 사용한 a 태그에 그냥 href값으로 KAKAO_AUTH_URL을 전달한 상태인데, 만일 페이스북, 구글, 네이버, 깃허브와 같은 다른 API들을 사용하기 위해 map을 돌리는 객체마다 다른 href를 주려면 어떻게 접근해야 할지 궁금합니다.

hsuj86 commented on src/pages/Login/Redirect.js:
```diff
@@ -0,0 +1,83 @@
+
+  const getToken = () => {
+    fetch(`http://10.58.6.112:8000/users/kakao?code=${code}`, {
+      method: 'GET',
```
- GET 요청은 삭제 가능합니다.

hsuj86 commented on src/pages/Login/OAuth.js:
```diff
@@ -0,0 +1,4 @@
+const CLIENT_ID = 'a7e335dc3c39abde914b9eca0f7ecab0';
+const REDIRECT_URI = 'http://localhost:3000/oauth/callback/kakao';
+
+export const KAKAO_AUTH_URL = `https://kauth.kakao.com/oauth/authorize?client_id=${CLIENT_ID}&redirect_uri=${REDIRECT_URI}&response_type=code`;
```
- 질문 주신 것에 대해 답변 드리면 
- export const NAVER_AUTH_URL = `url..`; 등과 같이 추가하시고 export 하신 후
- Login.js에서 KAKAO_AUTH_URL을 import 하신 것 처럼 다른 URL도 import 해주시고
- map 함수에 사용되는 데이터에 ex) url : KAKAO_AUTH_URL 같이 넣어주시면 될 것 같습니다.

hsuj86 reviewed (commented):
- 고생하셨습니다! 형겸님! 리뷰 확인해 주세요.

Kyeom1997 committed:
DELETE: GET Method 삭제

Modify: Conflict Resolve

ADD: Github Login API 사용

1. Github Oauth 사용해서 Redriect URI 및 Client Id OAuth.js에 추가
2. GithubRedirect.js 컴포넌트 생성

ADD: Github, Naver Redirect_URI 구현

ADD: .env 파일 생성

1. 각 API 별 client_id .env 파일에 분리
2. OAuth.js의 URI에 연결

ADD: config.js 파일 생성

1. BASE_URL 변수에 백엔드 통신 ip 할당
2. 사용하는 파일들에 import해서 fetch

hsuj86 commented:
- Conflict 해결해 주세요.
- 추가 기능 구현 다 되시면 라벨 변경해 주세요. 👍🏻

Kyeom1997 commented:
> * Conflict 해결해 주세요.
> * 추가 기능 구현 다 되시면 라벨 변경해 주세요. 👍🏻

---------------------------------------------

네이버, 페이스북, 구글, 깃허브 소셜 로그인까지 구현하면 라벨 변경하겠습니다!!

Kyeom1997 commented:
socialLogin Branch에서 config.js 파일이랑 .env 파일 생성해서 push 완료했습니다!

sstaar91 reviewed (commented):
형겸님 수고하셨습니다!
